packages/video_player/video_player_web/lib/video_player_web.dart

```diff
@@ -66,10 +66,13 @@ class VideoPlayerPlugin extends VideoPlayerPlatform {
     final int textureId = _textureCounter;
     _textureCounter++;
 
-    Uri uri;
+    String uri;
     switch (dataSource.sourceType) {
       case DataSourceType.network:
-        uri = Uri.parse(dataSource.uri);
+        // Validate the incoming uri
+        Uri.parse(dataSource.uri);
```

Comment:
What do we do with the result of this call?

Reply:
I think this was only added originally to *throw* if the passed in dataSource.uri was not valid, so I left it there only to "validate" the uri, since now we pass dataSource.uri directly.

Reply:
I'm not sure if it helps real-world apps, but Dart's `Uri` implementation is a spec-compliant implementation that's implemented in userspace. Last time I checked it was 18KB big. If this is for validation only, could it be wrapped in an `assert`?

Reply:
Tested this, and the plugin is able to report the "malformedness" of the URL to the user:

```
Error: PlatformException(MEDIA_ERR_SRC_NOT_SUPPORTED, MEDIA_ELEMENT_ERROR: Format error, The video
has been found to be unsuitable (missing or in a format not supported by your browser).)
```

I'll remove the Uri parsing to save a few kb. Thanks for bringing this up!